scout/adapter/mongo/case.py

```diff
@@ -555,6 +557,60 @@ def delete_case(self, case_id=None, institute_id=None, display_name=None):
         result = self.case_collection.delete_one(query)
         return result
 
+    def check_existing_data(self, case_obj, existing_case, institute_obj, update, keep_actions):
+        """Make sure data from case to be loaded/reuploaded conforms to case data already saved in database.
+           Return eventual evaluated variants to be propagated to the updated case if keep_actions is True
+
+        Args:
+            case_obj(dict): case dictionary to be loaded/reuploaded
+            existing_case(dict): a case with same _id or same display_name and institute_id as case_obj
+            institute_obj(dict): institute dictionary
+            update(bool): If existing case should be updated
+            keep_actions(bool): If old evaluated variants should be kept when case is updated
+
+        returns:
+            previous_evaluated_variants(list): list of variants evaluated in previous case
+        """
+
+        if existing_case is None:
+            return
+
+        # Avoid saving a new case in database with a display_name that already exists for another case from the same customer
+        if existing_case["_id"] != case_obj["_id"]:
+            raise IntegrityError(
+                "A case with display name %s already exists in database for this customer"
+                % case_obj["display_name"]
+            )
+
+        if existing_case and not update:
+            raise IntegrityError("Case %s already exists in database" % case_obj["_id"])
+
+        # Enforce same display name for updated case as existing case
+        if case_obj["display_name"] != existing_case["display_name"]:
+            raise IntegrityError("Updated case name doesn't match existing case name.")
+
+        # Check that individuals from updated case match individuals from existing case in ID, name and affected status
+        existing_case_inds = set(
+            [
+                (ind["individual_id"], ind["display_name"], ind["phenotype"])
+                for ind in existing_case.get("individuals")
+            ]
+        )
+        case_inds = set(
+            [
+                (ind["individual_id"], ind["display_name"], ind["phenotype"])
+                for ind in case_obj.get("individuals")
+            ]
+        )
+        if existing_case_inds != case_inds:
```

Comment:
The other integrity errors will print a hint to which parameter doesn't match. Might be a good idea to do it here as well?